src/components/implementation/nicmgr/dpdk/init.c

```diff
@@ -82,28 +157,35 @@ process_rx_packets(cos_portid_t port_id, char** rx_pkts, uint16_t nb_pkts)
 
 		if (htons(eth->ether_type) == 0x0800) {
 			iph	= (struct ip_hdr *)((char *)eth + sizeof(struct eth_hdr));
+			if (unlikely(iph->proto != UDP_PROTO)) {
+				cos_free_packet(buf.pkt);
+				rx_enqueued_miss++;
+				continue;
+			}
 			port	= (struct tcp_udp_port *)((char *)eth + sizeof(struct eth_hdr) + iph->ihl * 4);
 
-			session = find_session(iph->dst_addr, port->dst_port);
-			if (session == NULL) continue;
+			session = cos_hash_lookup(port->dst_port);
+			if (unlikely(session == NULL)) {
+				cos_free_packet(rx_pkts[i]);
+				continue;
+			}
 
 			buf.pkt = rx_pkts[i];
-			if (!pkt_ring_buf_enqueue(&session->pkt_ring_buf, &buf)){
+			if (unlikely(!pkt_ring_buf_enqueue(&session->pkt_ring_buf, &buf))){
 				cos_free_packet(buf.pkt);
+				rx_enqueued_miss++;
```

Comment:
Consider that these ++s might be faas so that it works across cores. Maybe I missed it, and the counters are TLS.